### PR TITLE
NP-46408 Add default status 'pending' on status filter on task page

### PR DESCRIPTION
**Pull request by joachimjorgensen**
# Description

Set the default value for the status filter on the task page to "Pending" ("Behandles"). Should not set default value on my-page.

Will also clean up the technical debt of "NP-46271 Ansvarsområde" after last sprint, where url params were set directly in the useQuery-function. This is now fixed by using useMemo and useEffect instead. Had to fix this to get the status filter working because of a race condition spawned by the Promise in useQuery.

Default value for curator filter in a different PR.

Link to Jira issue: https://unit.atlassian.net/browse/NP-46408

# Checklist

## Required

- [x] The changes are working as expected
- [x] The changes are tested OK for both desktop and mobile screens
- [x] The changes are tested OK for a11y (with [axe](https://www.deque.com/axe/devtools/), [lighthouse](https://developer.chrome.com/docs/lighthouse/), or similar)
- [x] Interactive elements have data-testids
- [x] I have done a QA of my own changes

## Optional

- [x] Solution is verified by PO/designer

### CODE REVIEW

File: src/components/TicketStatusFilter.tsx [hunk 1]
```diff
@@ -1,27 +1,34 @@
...
+import { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useHistory } from 'react-router-dom';
 import { TicketSearchParam } from '../api/searchApi';
 import { TicketStatus, ticketStatusValues } from '../types/publication_types/ticket.types';
 import { dataTestId } from '../utils/dataTestIds';
 
-export const TicketStatusFilter = () => {
+interface TicketStatusFilterProps {
+  isOnTasksPage: boolean;
```
Comment (elkebab) [inline src/components/TicketStatusFilter.tsx:10]: like greit å bare sende inn default valgte statuser fra parent her?
```suggestion
  defaultStatusFilter: TicketStatus[];
```

File: src/components/AreaOfResponsibiltySelector.tsx [hunk 2]
```diff
@@ -49,16 +50,19 @@ export const AreaOfResponsibilitySelector = () => {
...
-        searchParams.set(TicketSearchParam.ViewingScope, organizationOptions.map((org) => org.id).join(','));
-        history.push({ search: searchParams.toString() });
-      }
-      return organizationOptions;
+      return buildOrganizationOptions(organizations);
     },
     meta: { errorMessage: t('feedback.error.get_institution') },
   });
-  const organizationOptions = organizationQuery.data ?? [];
+  const organizationOptions = useMemo(() => organizationQuery.data ?? [], [organizationQuery]);
```
Comment (elkebab) [inline src/components/AreaOfResponsibiltySelector.tsx:57]: god praksis å være mest mulig spesifikk, for å unngå å gjøre det oftere enn nødvendig. I dette tilfelle har det neppe noen praktisk betydning, but still
```suggestion
  const organizationOptions = useMemo(() => organizationQuery.data ?? [], [organizationQuery.data]);
```

File: src/components/TicketStatusFilter.tsx [hunk 3]
```diff
@@ -1,27 +1,34 @@
...
   const { t } = useTranslation();
   const history = useHistory();
-  const searchParams = new URLSearchParams(history.location.search);
-  const selectedStatuses = (searchParams.get(TicketSearchParam.Status)?.split(',') ??
-    ticketStatusValues) as TicketStatus[];
 
-  const handleChange = (event: SelectChangeEvent<string[]>) => {
-    const newSelectedStatuses = event.target.value as TicketStatus[];
+  const defaultStatusFilter = (isOnTasksPage ? ['Pending'] : ticketStatusValues) as TicketStatus[];
+  const [selectedStatuses, setSelectedStatuses] = useState(defaultStatusFilter);
```
Comment (elkebab) [inline src/components/TicketStatusFilter.tsx:18]: Trengs det en egen `useState` her? Lukter litt dobbeltlagring/skyggeføring siden vi har status i URLen.
Kunne noe sånt gjort samme nytten? (tolk dette som pseudokode)
```
const statusesFromParams = searchParams.get(TicketSearchParam.Status)?.split(',')
const selectedStatuses = statusesFromParams.length > 0
  ? statusesFromParams
  : defaultStatuses
```

File: src/components/TicketStatusFilter.tsx (hunk 1 above)
Comment (joachimjorgensen) [reply src/components/TicketStatusFilter.tsx:10]: Lurt!

File: src/components/AreaOfResponsibiltySelector.tsx [hunk 4]
```diff
@@ -49,16 +50,19 @@ export const AreaOfResponsibilitySelector = () => {
...
+
+  useEffect(() => {
+    const sp = new URLSearchParams(history.location.search);
+    if (!sp.get(TicketSearchParam.ViewingScope)) {
```
[the remark this answers is not in the document]
Comment (joachimjorgensen) [reply src/components/AreaOfResponsibiltySelector.tsx:61]: Prøvde, men da må jeg legge searchParams i useMemo, og da rendrer den altfor ofte... Mye det samme som vi slet med sist.

File: src/components/TicketStatusFilter.tsx (hunk 3 above)
Comment (joachimjorgensen) [reply src/components/TicketStatusFilter.tsx]: Tror jeg fikk den til å virke. Måtte ta i bruk en `useEffect()` for å sette param ved første render da.

**Comment by joachimjorgensen** [comment]
Abandoner denne da dette ble refaktorert i https://github.com/BIBSYSDEV/NVA-Frontend/pull/5352

**Pull request closed without merging**